test/hotspot/jtreg/compiler/arguments/TestCodeEntryAlignment.java

```diff
@@ -66,9 +66,10 @@ public static void shouldPass(String... args) throws IOException {
     }
 
     public static void driver() throws IOException {
-        for (int align = 16; align < 256; align *= 2) {
+        for (int align = 32; align <= 1024; align *= 2) {
```

Comment:
This wouldn't test for `-XX:CodeEntryAlignment=16`.
However, we used to find a bug with `-XX:CodeEntryAlignment=16`.
https://github.com/openjdk/jdk/pull/7485

Why not testing with `-XX:CodeEntryAlignment=16`?

Reply:
The only reason is because -XX:CodeCacheSegmentSize=16 gives an error.  If it's important to test for -XX:CodeEntryAlignment=16 then I'll have to rework the logic to be more clever.  I'll probably have to use WhiteBox APIs to read the default CodeCacheSegmentSize and only set it to values that are >= the default.  It's unfortunate that -XX:CodeEntryAlignment doesn't adjust the default value of CodeCacheSegmentSize automatically.

Reply:
Maybe we can add testing for `-XX:CodeEntryAlignment={512, 1024}` like this.
```
diff --git a/test/hotspot/jtreg/compiler/arguments/TestCodeEntryAlignment.java b/test/hotspot/jtreg/compiler/arguments/TestCodeEntryAlignment.java
index fd6c8ca..f07030e 100644
--- a/test/hotspot/jtreg/compiler/arguments/TestCodeEntryAlignment.java
+++ b/test/hotspot/jtreg/compiler/arguments/TestCodeEntryAlignment.java
@@ -72,6 +72,14 @@ public class TestCodeEntryAlignment {
                 "-XX:CodeEntryAlignment=" + align
             );
         }
+
+        for (int align = 512; align < 1024; align *= 2) {
+            shouldPass(
+                "-XX:+UnlockExperimentalVMOptions",
+                "-XX:CodeCacheSegmentSize=" + align,
+                "-XX:CodeEntryAlignment=" + align
+            );
+        }
     }
 
 }
```
What do you think?

Reply:
That sounds good.  Done.